tests/test_apigw.py

```diff
@@ -214,3 +216,22 @@ def test_rest_stage_update(self):
             self.assertEqual(m["loggingLevel"], "INFO")
             self.assertEqual(m["dataTraceEnabled"], True)
         self.assertTrue(found)
+
+    def test_rest_stage_delete(self):
+        session_factory = self.replay_flight_data("test_rest_stage_delete")
+        p = self.load_policy(
+            {
+                "name": "rest-stage-delete",
+                "resource": "rest-stage",
+                "filters": [{"type": "value", "key": "stageName", "value": "delete-test"}],
+                "actions": [{"type": "delete"}],
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        client = session_factory().client("apigateway")
+        with self.assertRaises(ClientError) as e:
+            stage = client.get_stage(
```

Comment:
@faizanfahmed small lint error here, you don't need to assign stage to the return value of get_stage